DOCS-12496 draft for mTLS (#53)
mulesoft/docs-anypoint-datagraph: modules/ROOT/pages/add-api-to-unified-schema.adoc

. Click *Next: Provide authentication*.
. On the *Configure Security* page, select the type of authentication on the GET endpoints for the API that you are adding:
+
* *No Auth*: Use if your API is public.
* *Basic Auth*: Use if you authenticate to your API using a username and password.
* *Pass-through*: Use if you authenticate to your API by passing authorization headers.
* *Client ID enforcement via headers*: Use if you authenticate to your API passing `client_id` and `client_secret` headers.
* *Client ID enforcement via query parameters*: Use if you authenticate to your API passing `client_id` and `client_secret` query params.
* *Custom*: Use if you authenticate to your API using custom header parameters and values.
. Still on the *Configure Security* page, optionally configure mutual authentication (mTLS) between the API and DataGraph. 
.. Select *Configure Keystore* and provide the following:
* A public certificate signed by your organization's “root” certificate authority (CA)
* The private key used to generate the certificate signing request (CSR)
* The password for the encrypted private key

> .. Optionally, select *Configure Truststore* and provide a link to the root CA.

Reviewer: the user will not configure mTLS as a whole, the keystore and the truststore are two separate containers that the user can optionally configure. So the user will be able to include either the keystore or the truststore, or both if needed. Also for the truststore, the user uploads files just as it would do for the keystore's public certificate (the user can upload one file with many certificates, or many files with one certificate, or a mix of both)

* If your API is managed by API Manager or if you have added the API’s instances in Exchange, select *Get an existing URL from Anypoint Platform*.
* If you prefer to manually provide your own implementation URL, select *Add a new URL*.
. Click *Next: Provide authentication*.
. On the *Configure Security* page, select the type of authentication on the GET endpoints for the API that you are adding:
+
* *No Auth*: Use if your API is public.
* *Basic Auth*: Use if you authenticate to your API using a username and password.
* *Pass-through*: Use if you authenticate to your API by passing authorization headers.
* *Client ID enforcement via headers*: Use if you authenticate to your API passing `client_id` and `client_secret` headers.
* *Client ID enforcement via query parameters*: Use if you authenticate to your API passing `client_id` and `client_secret` query params.
* *Custom*: Use if you authenticate to your API using custom header parameters and values.

> . Still on the *Configure Security* page, optionally add a keystore to configure mutual authentication (mTLS) between the API and DataGraph. 

Author: @tomasiturralde revised steps

Reviewer: looks good!

+
Select *Configure Keystore* and provide the following:
+
* A public certificate signed by your organization's “root” certificate authority (CA)
* The private key used to generate the certificate signing request (CSR)
* The password for the encrypted private key

> . Optionally, add a truststore to allow your API to be reached your unified schema.

Author: ```suggestion
. Optionally, add a truststore to allow your API to be reached by your unified schema.
```

> . Optionally, add a truststore to allow your API to be reached your unified schema.

Author: ```suggestion
. Optionally, add a truststore to allow your API to be reached by your unified schema.
```

. Select an authentication policy and complete any required fields.
. Click *Save*.

[[configure-mtls-for-an-api-added-to-the-unified-schema]]
== Configure mTLS for an API Added to the Unified Schema

If you need to configure mTLS for an API that you've added to the unified schema, you can do so on the *API details* page.

. Click *List of APIs added* and select the API that has the URL you want to edit.
. Click *API details*.

> . .. Select *Configure Keytore*, and provide the following:

Reviewer 2: ```suggestion
. .. Select *Configure Keystore* and provide the following:
```

* A public certificate signed by your organization's “root” certificate authority (CA).
* The private key used to generate the certificate signing request (CSR).

> * The password for the encrypted private key.

Reviewer 2: no periods

> .. Optionally, select *Configure Truststore* and provide a link to the root CA.

Reviewer: Same as with the add api. I would also add that the user can remove both the keystore and the truststore if the api had previously added ones. however the user will not be able to modify the keystore and truststore per se, it will need to remove the current keystore/truststore and add a new one. it can be done within the same action though

. Select an authentication policy and complete any required fields.
. Click *Save*.

> [[configure-mtls-for-an-api-added-to-the-unified-schema]]

Author: @tomasiturralde revised steps

Reviewer: looks good

== Configure mTLS for an API Added to the Unified Schema

If you need to configure mTLS for an API that you've added to the unified schema, you can do so on the *API details* page by adding a keystore. 

> You can also add a truststore to allow your API to reach your unified schema.

Author: ```suggestion
You can also add a truststore to allow your API to be reached by your unified schema.
```

> You can also add a truststore to allow your API to be reached your unified schema.

Reviewer 3: missing word?

> You can also add a truststore to allow your API to be reached your unified schema.

Author: ```suggestion
You can also add a truststore to allow your API to reach your unified schema.
```

> You can also add a truststore to allow your API to be reached your unified schema.

Author: ```suggestion
You can also add a truststore to allow your API to be reached by your unified schema.
```

[NOTE]
If the API already has a keystore or truststore, you can remove either and add new ones on the *API details* page. To do so, click *Remove and add a new one*.

. Click *List of APIs added* and select the API that has the URL you want to edit.
. Click *API details*.

> . Select *Configure Keystore* and provide the following:

Reviewer 3: Does this sublist work? (I usually add the + and ** for a sublist.)

Author: they do work
